File: lab3_medidas_cent_var.py
# Neste laboratorio estaremos trabalhando com as ferramentas disponiveis de Python para medidas de centralidade

import numpy as np
from scipy import stats


jogadores = [40000, 18000, 12000, 250000, 30000, 140000, 300000, 40000, 800000]
# Vamos calcular a media
media_salarios = np.mean(jogadores)
print('O valor da media salarial dos jogadores e: ', media_salarios)
# Calculo da mediana
mediana_salarios = np.median(jogadores)
print('\nO valor da mediana salarial dos jogadores e: ', mediana_salarios)
# Calculo dos quartis => note que temos que definir quais os quartis que queremos calcular
quartis_salarios = np.quantile(jogadores, [0, 0.25, 0.5, 0.75, 1])
print('\nVisao geral dos quartis: ', quartis_salarios)
# Calculo do desvio padrao do salario dos jogadores => observe que o resultado e diferente do obtido em R
dpadrao_salarios = np.std(jogadores)
print('\nO desvio padrao do salario dos jogadores e: ', dpadrao_salarios)
# Calculo do desvio padrao do salario dos jogadores => observe os parametros necessarios para obter o mesmo resultado de R (em R o parametro ddof utiliza no denominador (n-1) e no Python apenas (n) =>> para tornar os resultados equivalentes voce devera corrigir isso)
dpadrao_ddof_salarios = np.std(jogadores, ddof=1)
print('\nO desvio padrao corrigido para R do salario dos jogadores e: ', dpadrao_ddof_salarios)
# Uma forma rapida de visualizar todos os parametros de centralidade de uma base de dados e com o uso do metodo describe()
resumo_salarios = stats.describe(jogadores)
print('\nO resumo das medidas de centralidade do salario dos jogadores e: \n\n', resumo_salarios)

File: lab3_medidas_cent_var.R
# Title     : Medidas de Centralidade e de Variabilidade
# Objective : Apresentar as principais ferramentas em R para medidas de centralidade e variabilidade
# Created by: accol
# Created on: 21/05/2020
# Medidas de Centralidade em R >>= Funcoes no R:
# quantile() => para o calculo dos quartis
# sd() => para o calculo do desvio padrao
# var() => para o calculo da variancia
# mean() => para o calculo da media
# median() => para o calculo da mediana
# summary() => visualizacao rapida da base de dados

# Para este laboratorio estaremos gerando alguns dados para analise

jogadores = c(40000, 18000, 12000, 250000, 30000, 140000, 300000, 40000, 800000)
media_salarios = mean(jogadores) # Calculo da media
mediana_salarios = median(jogadores) # Calculo da mediana
quartis_salarios = quantile(jogadores) # Calculo dos quartis
print(media_salarios)
print(mediana_salarios)
print(quartis_salarios)
# Para visualizar um quartil especifico, basta o seguinte <$> Atencao => indices em R iniciam em 1
print(quartis_salarios[1])
desviop_salarios = sd(jogadores) # Calcula o desvio padrao
print(desviop_salarios)
# Caso deseje visualizar rapidamente um resumo de sua base de dados, podera utilizar a funcao summary()
print('Resumo da base de dados:')
resumo_salarios = summary(jogadores)
print(resumo_salarios)
